great_expectations/data_context/types/base.py

```diff
@@ -2306,6 +2313,7 @@ class DataContextConfig(BaseYamlConfig):
 
     def __init__(  # noqa: C901, PLR0912, PLR0913, PLR0915
         self,
+        batch_configs: Optional[Dict] = None,
```

Comment:
Do we want to strictly type the `dict` here?

Reply:
I'm on the fence and can be swayed. My thought is that ideally this would be using pydantic for validation (which will end up happening later on when we load the model itself. So this feels like a least-effort-possible intermediary step while we are using marshmallow. This is what we do for fluent datasources, and I was mostly leaning on that as prior art.

Reply:
Yeah I'm okay with this